src/animated.hpp

```diff
@@ -119,12 +119,28 @@ class animated
 private:
 	struct frame
 	{
-		// Represents the timestamp of the frame start
-		std::chrono::milliseconds duration_;
 		T value_;
+
+		// Represents the timestamp of the frame start
 		std::chrono::milliseconds start_time_;
+		std::chrono::milliseconds duration_;
 	};
 
+	static auto get_frame_begin_time(const frame& frame)
+	{
+		return frame.start_time_;
+	}
+
+	static auto get_frame_end_time(const frame& frame)
+	{
+		return frame.start_time_ + frame.duration_;
+	}
+
+	static auto get_frame_duration(const frame& frame)
```

Comment:
I don't see what the `get_frame_begin_time` and `get_frame_duration` functions add beyond directly accessing the structure members. Also, I don't see why they're static instead of member functions.

Reply:
Just decouples the begin/end/duration getters from the current frame (`get_current_frame_begin_time`, etc). I had them as static to keep `frame` as a PoD struct, but yeah, no reason they can't be members.

Reply:
Whatever this is supposed to improve it should not be mixed in the same commit with actual changes. Same thing with shuffling the `frame` members around.